sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/ShuffleExchangeExec.scala

```diff
@@ -77,18 +92,29 @@ trait ShuffleExchangeLike extends Exchange {
   def runtimeStatistics: Statistics
 }
 
+// Describes where the shuffle operator comes from.
+object ShuffleOrigin extends Enumeration {
```

Comment:
Can we use seal trait/abstract class, so we type won't have a trailing `.Value`?